crates/ruma-common/src/events/room/message.rs

```diff
@@ -221,7 +221,11 @@ pub enum MessageType {
 }
 
 impl MessageType {
-    /// Creates a new `MessageType` with the given `msgtype` string and data.
+    /// Creates a new `MessageType`.
+    ///
+    /// The `msgtype` and `body` are required fields as defined by [the `m.room.message` spec](https://spec.matrix.org/v1.2/client-server-api/#mroommessage).
+    /// Additionally it's possible to add arbitrary key/value pairs to the event content through the
+    /// `data` map.
```

Comment:
Can you move this below the other paragraph and add sth. like

```suggestion
    /// The `msgtype` and `body` are required fields as defined by [the `m.room.message` spec](https://spec.matrix.org/v1.2/client-server-api/#mroommessage).
    /// Additionally it's possible to add arbitrary key/value pairs to the event content for custom events through the
    /// `data` map.
```